ResearchJournal.tex

```diff
@@ -0,0 +1,101 @@
+
+When maintaining the open nodes of an A* search, it is better to use an array of stacks than a priority queue. If $N$ is the number of elements in the priority queue, the insertion and the extraction of an element both take $O(log N)$ which is faster than a constant time for inserting and a linear time for extracting as in the list implementation \cite{cazenave2006optimizations}.  Correctly maintaining the priority queue is key for optimization, allowing a lazier initialisation for the node cache; it is more efficient to use an array of the size of the map as the game map already fits in memory \cite{cazenave2006optimizations}.
+
+\subsection*{Variations}
+
+Common optimisations are to implement hierarchy (HPA*) with space abstractions or by using a navmesh \cite{cui2011based}.  HPA* breaks down the map into sections and is significantly faster than low level A* \cite{botea2004near}, speeding up a search by replacing it with a series of smaller searches \cite{botea2013pathfinding} gradually refined into a low level solution.  It is also possible to build more than one level of abstraction for HPA* \cite {jansen2007hpa}. By precomputing ideal routes across these larger sections; encouraging the algorithm to expand nodes from the perimeter rather than the interior \cite{harabor2012fast}, or node pruning with jump points \cite{harabor2011online}, symmetries (identical length paths where the only difference between them is the order in which moves appear \cite{harabor2012fast}) can be drastically reduced, trading off the real time CPU resources with optimality of the solution. 
+
+Iterative A* is a space-efficient version of A*, but it suffers from cycles in the search space (the price for using no storage), and repeated visits to states (the overhead of iterative deepening) \cite{bjornsson2005fringe}.
+
+Block A* pre-computes a database of all possible block topologies.  For each block, optimal distances between pairs of boundary cells are stored.  At runtime, Block A* expands an entire block at a time, as opposed to one individual cell \cite{botea2013pathfinding}
+
+\subsection*{Always with the triangles, your solution to everything is triangles}
```

Comment:
Not sure about this title but I'll let it slide! Although many navigation mesh systems allow for polygons with more than three sides, so the solution isn't always triangles